### PR TITLE
Add apps.darwin-{un,}installer and make installer defaultApp.

### PR DESCRIPTION
**Pull request by jsoo1**
#398

### CODE REVIEW

**Comment by LnL7** [comment]
This is the regular installer which will setup and create a `configuration.nix`, for flakes I would assume you want a different process.

**Comment by jsoo1** [comment]
> This is the regular installer which will setup and create a `configuration.nix`, for flakes I would assume you want a different process.

Yes I think you are right. I almost imagine that `darwin-rebuild` would be enough, no?

**Comment by Pegasust** [comment]
I also really like this idea, what are currently the blockers on this?

**Comment by jsoo1** [comment]
Mostly I haven't rebased this work and things changed a lot but nothing is fundamentally blocking it aside from that.

File: flake.nix [hunk 1]
```diff
@@ -2,7 +2,42 @@
...
-  outputs = { self, nixpkgs }: {
+  inputs = {
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, flake-utils, nixpkgs }:
+    let
+      darwinSystems =
+        builtins.filter
+          (sys: builtins.isList (builtins.match "^[[:alnum:]_]+-darwin$" sys))
```
Comment (Enzime) [inline flake.nix:13]: Might be simpler to use `hasSuffix` here

Comment (Enzime) [reply flake.nix:13]: Alternatively, you could hardcode it to `aarch64-darwin` and `x86_64-darwin`, as currently it also includes `armv7a-darwin` and `i686-darwin`

File: flake.nix [hunk 2]
```diff
@@ -2,7 +2,42 @@
...
+          flake-utils.lib.allSystems;
+
+      portable = flake-utils.lib.eachSystem darwinSystems
+        (system:
+          let pkgs = import nixpkgs { inherit system; };
+
+          in rec {
+            defaultApp = apps.darwin-installer;
+
+            apps = {
```
[the remark this answers is not in the document]
Comment (jsoo1) [reply flake.nix:23]: I probably won't have the time to finish this pr.  Would you be interested in taking it over?

[the remark this answers is not in the document]
Comment (Enzime) [reply flake.nix:23]: Yeah I'm happy to take over the PR :+1:

[the remark this answers is not in the document]
Comment (jsoo1) [reply flake.nix:23]: Ok. I think I'll have to close it if you want to rebase it and do the work.

**Comment by emilazy** [comment]
Closing in favour of https://github.com/LnL7/nix-darwin/pull/687. Thank you for doing the pioneering work on this!

**Comment by jsoo1** [comment]
> Closing in favour of #687. Thank you for doing the pioneering work on this!

I'm glad to see it picked up :)

**Pull request closed without merging**